### PR TITLE
 Code refactoring of GetRoutes + getting Tours by Tourname

### PR DESCRIPTION
**Pull request by lauravikanis**
Refactor MongoDB getRoutes, get Tours by Tournames (useLocation) 
-> you can now get Locationentries and Tour description for variable Tours

### CODE REVIEW

**Review by boldCM — APPROVED** [review]
lgtm🎉

File: client/src/App.js [hunk 1]
```diff
@@ -32,7 +32,7 @@ function App() {
           <Route path="/choice">
             <Choice />
           </Route>
-          <Route path="/tour">
+          <Route exact path="/tour/:tourname">
```
Comment (domko95) [inline client/src/App.js:35]: Are you sure you need an exact in front of a variable path?
```suggestion
          <Route path="/tour/:tourname">
```

File: server.js [hunk 2]
```diff
@@ -19,10 +17,10 @@ app.use(
   express.static(path.join(__dirname, "client/storybook-static"))
 );
 
-app.get("/api/locations/:tour", async (req, res) => {
-  const { tour } = req.params;
+app.get("/api/locations/name/:name", async (req, res) => {
+  const { name } = req.params;
```
Comment (domko95) [inline server.js:21]: If you want to search through your locations for a specific name, you could use queries instead.
```suggestion
app.get("/api/locations", async (req, res) => {
  const { name } = req.query;
```

In that case you would need to change your route in your api/locations.js.

File: client/src/api/locations.js [hunk 3]
```diff
@@ -1,25 +1,19 @@
-export async function getLocationNamebyTour() {
-  const response = await fetch(`/api/locations/electronic`);
-
+export async function getLocationByName(name) {
+  const response = await fetch(`/api/locations/name/${name}`);
```
Comment (domko95) [inline client/src/api/locations.js:2]: According to another comment further down, if you would want to adapt that, you would need to change it here as well.
```suggestion
export async function getLocationByName(name) {
  const response = await fetch(`/api/locations?name=${name}`);
```

**Review by domko95 — APPROVED** [review]
LGTM 🚀

File: client/src/App.js (hunk 1 above)
Comment (lauravikanis) [reply client/src/App.js:35]: changed this to a path which is notexact

File: server.js (hunk 2 above)
Comment (lauravikanis) [reply server.js:21]: Implemented 🙌 THX for suggesting this.

File: client/src/api/locations.js (hunk 3 above)
Comment (lauravikanis) [reply client/src/api/locations.js:2]: Implemented 🙌 THX for suggesting this.

**Review by SvLowman — COMMENTED** [review]
👍

**Pull request merged**